encrypted-media-respec.html

```diff
@@ -694,15 +694,23 @@
                   </dd>
                   <dt><a def-id="requirement-optional"></a></dt>
                   <dd>
-                    <p>Continue.<!-- Will be updated below. --></p>
+                    <p>If <a def-id="distinctive-identifiers"></a> are not allowed according to <var>restrictions</var>:</p>
+                    <ol>
+                      <li>
+                        <p>Set the <a def-id="option-distinctiveIdentifier"></a> member of <var>accumulated configuration</var> to <a def-id="requirement-not-allowed"></a></p>
+                      </li>
+                      <li>
+                        <p>If the <var>implementation</var> requires a <a def-id="distinctive-identifier"></a> in combination with <var>accumulated configuration</var> and <var>restrictions</var>, return <code>NotSupported</code>.</p>
```

Comment:
Actually, that doesn't work because we are modifying a different value than was checked. One solution is to use a local variable per my comment at line 713.

Alternatively, we could avoid this complexity by extracting this into a step before the "switch statement." We would need to restore the "Continue..." and check for "optional" twice, but it might be less complex.

For example:
* `Let <var>distinctive identifier requirement</var> be the value of <var>candidate configuration</var>'s <a def-id="option-distinctiveIdentifier"></a> member.`
* `If  <var>distinctive identifier requirement</var> is <a def-id="requirement-optional"></a> and <a def-id="distinctive-identifiers"></a> are not allowed according to <var>restrictions</var>, let <var>distinctive identifier requirement</var> be <a def-id="requirement-not-allowed"></a>.`
* Execute steps before this commit...